### ISSUE TITLE
firefox-esr segfaults when playing media

### ISSUE THREAD

**Issue by lukegb** [NixOS/nixpkgs#121408]
**Describe the bug**
https://hydra.nixos.org/build/142427422/nixlog/2

```
machine # [   87.445436] MediaDe~hine #1[1250]: segfault at 0 ip 00007fce8cf2fca9 sp 00007fce6af88e00 error 4 in libmozsandbox.so[7fce8cf25000+f000]
machine # [   87.449473] Code: 74 13 89 cf 81 e7 00 10 00 00 74 09 80 3a 00 0f 84 cd 00 00 00 48 c7 c3 da ff ff ff 48 85 f6 0f 84 b4 01 00 00 83 f8 9c 74 09 <80> 3a 2f 0f 85 f5 00 00 00 89 cf 81 e7 ff f6 ff ff 0f 84 cf 00 00
```

**To Reproduce**
1. Run `nix-build ./nixos/release.nix -A tests.firefox-esr.x86_64-linux`

**Expected behavior**

The test should pass.

**Screenshots**

It hangs forever waiting for audio to happen.

**Additional context**

**Notify maintainers**

Recent changes:
@petabyteboy for Rust #121290 
@mweinelt as maintainer

**Metadata**
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
  - firefox-esr
# a list of nixos modules affected by the problem
module:
```

**Comment by mweinelt** [comment]
TL;DR: YouTube tabs are crashing after 903e23ad362ed19e95eaf7765af719db551dfbbe. Works after reverting the commit.

**Comment by petabyteboy** [comment]
Thanks for the report! I can confirm #121290 causes this and I'll revert it.

**Comment by lukegb** [comment]
Wow, thanks both for the super speedy response. :)

**Comment by petabyteboy** [comment]
Nah, that change shouldn't have been merged in the first place.

**Comment by kvtb** [comment]
that's because `clang` and `rust` use different `llvm` versions

**Comment by petabyteboy** [comment]
No, that was not the issue. In May, when this PR was made, Rust was not using LLVM 12 yet.
